src/datasets/builder.py

```diff
@@ -314,36 +318,30 @@ def __init__(
             self.info.features = features
 
         # Prepare data dirs:
-        # cache_dir can be a remote bucket on GCS or S3 (when using BeamBasedBuilder for distributed data processing)
-        self._cache_dir_root = str(cache_dir or config.HF_DATASETS_CACHE)
-        self._cache_dir_root = (
-            self._cache_dir_root if is_remote_url(self._cache_dir_root) else os.path.expanduser(self._cache_dir_root)
-        )
-        path_join = posixpath.join if is_remote_url(self._cache_dir_root) else os.path.join
+        self._cache_dir_root = str(cache_dir) or os.path.expanduser(config.HF_DATASETS_CACHE)
+        self._cache_dir = self._build_cache_dir()
         self._cache_downloaded_dir = (
-            path_join(self._cache_dir_root, config.DOWNLOADED_DATASETS_DIR)
+            os.path.join(self._cache_dir_root, config.DOWNLOADED_DATASETS_DIR)
             if cache_dir
-            else str(config.DOWNLOADED_DATASETS_PATH)
+            else os.path.expanduser(config.DOWNLOADED_DATASETS_PATH)
         )
-        self._cache_downloaded_dir = (
-            self._cache_downloaded_dir
-            if is_remote_url(self._cache_downloaded_dir)
-            else os.path.expanduser(self._cache_downloaded_dir)
-        )
-        self._cache_dir = self._build_cache_dir()
-        if not is_remote_url(self._cache_dir_root):
-            os.makedirs(self._cache_dir_root, exist_ok=True)
-            lock_path = os.path.join(self._cache_dir_root, self._cache_dir.replace(os.sep, "_") + ".lock")
-            with FileLock(lock_path):
-                if os.path.exists(self._cache_dir):  # check if data exist
-                    if len(os.listdir(self._cache_dir)) > 0:
-                        logger.info("Overwrite dataset info from restored data version.")
-                        self.info = DatasetInfo.from_directory(self._cache_dir)
-                    else:  # dir exists but no data, remove the empty dir as data aren't available anymore
-                        logger.warning(
-                            f"Old caching folder {self._cache_dir} for dataset {self.name} exists but not data were found. Removing it. "
-                        )
-                        os.rmdir(self._cache_dir)
```

Comment:
I see you are reverting the changes introduced by:
- #4347

Some concerns:
- What about if the user sets a remote `config.HF_DATASETS_CACHE` or `config.DOWNLOADED_DATASETS_DIR`?
- The `FileLock` was raising errors when working with a remote cache dir (BeamBasedBuilder)

Reply:
Good catch thanks !

The FileLock is should only be applied when the output directory is a local directory indeed